runtime/serde/serde-json/TESTING.md

```diff
@@ -0,0 +1,178 @@
+How to run JSONTestSuite against serde-json deserialize
+========================================================
+
+When making changes to the lexer it is a good idea to run the
+changes against the [JSONTestSuite](https://github.com/nst/JSONTestSuite) and manually examine the test results.
+
+### How to setup the JSONTestSuite
```

Comment:
JSONTestSuite is a bit interesting/onerous to setup and use. It does not lend itself to this kind of integration and I see it as an operational cost not worth paying.

Reply:
In this line of thinking, can this code as-is serve as a general purpose JSON parser?  If so perhaps it would make sense to break out as a separate dependency at some point in the future such that JSONTestSuite could run against it as the others they have in their repo.  Obviously not something for this PR.

Reply:
Unless we are planning to support it as a general purpose parser I would think it probably isn't in our interest (or anyone else) to do that. Perhaps we fork it and set it up so that it's easier to run...